src/tdb_encode_model.c

```diff
@@ -411,11 +412,13 @@ tdb_error make_grams(FILE *grouped,
     TDB_TIMER_END("encode_model/find_candidates")
 
     /* collect frequencies of *all* occurring bigrams of candidate unigrams */
-    TDB_TIMER_START
-    ret = event_fold(all_bigrams, grouped, num_events, items, num_fields, &g);
-    if (ret)
-        goto done;
-    TDB_TIMER_END("encode_model/all_bigrams")
+    if (!no_bigrams) {
```

Comment:
I don't remember if I specifically checked between these two functions but I do know that this particular change I made has quite big impact.

I wasn't sure if `choose_grams` is something that can be skipped. Maybe I will try removing both/or trying to remove `choose_grams` only.